src/visualizers/Chaos.ts

```diff
@@ -1,429 +1,804 @@
+        }
+    }
+
+    // Draws the dots between start and end INCLUSIVE
+    drawVertices(start: number, end: number) {
+        const sketch = this.sketch
+        for (let currWalker = 0; currWalker < this.walkers; currWalker++) {
+            // for each walker we look in dotsIndices
+            // to check if we are between
+            // start and end; this could be empty
+            const startArrayIndex = math.max(
+                this.dotsIndices[currWalker].findIndex(
+                    n => n >= BigInt(start)
+                ),
+                0
             )
-            for (let c = 0; c < this.corners; c++) {
-                const label = cornersLabels[c]
-                this.sketch.text(String(c), label.x, label.y)
+            let lastPos = this.dots[currWalker][startArrayIndex]
+            for (
+                // avoid i=0, dummy position
+                let i = math.max(1, startArrayIndex + 1);
+                this.dotsIndices[currWalker][i] <= end;
+                i++
+            ) {
+                const pos = this.dots[currWalker][i]
+                if (pos.x !== lastPos.x || pos.y !== lastPos.y) {
+                    sketch.fill(this.dotsColors[currWalker][i])
+                    const size = this.dotsSizes[currWalker][i]
+                    //sketch.push()
+                    //sketch.translate(lastPos.x, lastPos.y)
+                    //sketch.circle(pos.x, pos.y, this.dotsSizes[currWalker][i])
+                    //sketch.rect(pos.x, pos.y, size, size)
+                    this.polygon(pos.x, pos.y, size)
+                    //sketch.pop()
+                }
+                lastPos = pos
             }
         }
+    }
+    // from p5.js docs examples
+    polygon(x: number, y: number, radius: number) {
+        const angle = this.sketch.TWO_PI / this.sides
+        this.sketch.beginShape()
+        for (let a = 0; a < this.sketch.TWO_PI; a += angle) {
+            const sx = x + this.sketch.cos(a) * radius
+            const sy = y + this.sketch.sin(a) * radius
+            this.sketch.vertex(sx, sy)
+        }
+        this.sketch.endShape(this.sketch.CLOSE)
+    }
+    mouseWheel(event: WheelEvent) {
+        super.mouseWheel(event)
+        this.cursor = 0 // make sure we redraw
+    }
 
-        // no stroke (in particular, no outline on circles)
-        this.sketch.strokeWeight(0)
+    // how many dots do we have stored?
+    currentLength() {
+        let currentLength = -this.walkers // ignore dummy dots
+        for (let currWalker = 0; currWalker < this.walkers; currWalker++) {
+            currentLength += this.dots[currWalker].length
+        }
+        return currentLength
     }
 
     draw() {
+        // check if we are zoom/panning, redraw if so
+        // p5 is doing something terrible right here
+        // its terribleness is proportional to the
+        // size of the stored dot data
+        // sometimes image degrades during this pause
```

Comment:
Are all of the above comments still true? Or did the fix about creating geometry objects more or less take care of them?

Reply:
will make another comment soon